### ISSUE TITLE
field mapping of additional types like dates

### ISSUE THREAD

**Issue by t0bst4r** [PgBulkInsert/PgBulkInsert#58]
Hi again,

at the moment you are only mapping simple types in function JpaMapping#mapField.
First of all it would be nice to have some more mappings like LocalDate etc.

But for more data types you could just create a simple annotation like `@PgJpaDataType(value=DataType)` which can be parsed like the Enumerated-annotation in line 97 of JpaMapping.
Maybe this annotation could be used for other mappings, too, instead of jpa-only.

Then the developer could provide a custom type mapping instead of passing in a `postgresColumnMapping`. I think the annotation is more usable if combined with Jpa.
The `postgresColumnMapping`-map needs the column/field name as string again which we dont want to specify outside of the jpa data class.

Another option would be to use the `@Column(columnDefinition=String)` option. But this would need a bit more parsing to get the correct DataType enum value.

**Comment by bytefish** [comment]
Yes, it's a good idea to provide an additional annotation. I will create an initial version and fix the bug in the other ticket, too. 😎

**Comment by t0bst4r** [comment]
Perfect, thank you!
Another good option could be an additional or alternative Type-Map in the constructor of JpaMapping:

```java
public JpaMapping(Class<T> entityClass, Map<Class<*>, DataType> typeMapping) {
}
```

This map could be used to directly map java types/classes to your DataType enum --> in general.
If combined with a factory pattern in our app it could be a general setting for global data type mappings. This could be helpful for your if-else clauses, too.
You could do this in the same way so that this part of the code will be a bit shorter.
Maybe I'll create a pull request tomorrow.

**Comment by bytefish** [comment]
I have implemented a first version and wanted to upload it to Maven as 5.1.0-alpha1... but... Central Maven repositories are currently down and I cannot push the build to the Staging Repository, see:

* https://status.maven.org/incidents/rh8hfcm8h2kc

You can see how to use the Annotation:

https://github.com/bytefish/PgBulkInsert/blob/ad4761beaa360f1c534e2cf9c529bc91243f1433/PgBulkInsert/pgbulkinsert-jpa/src/test/java/de/bytefish/pgbulkinsert/test/jpa/annotations/PostgresDataTypeTests.java#L30


The Default Global Mappings for Data Types can be adjusted by passing a ``PostgresTypeMapping`` into the ``JpaMapping`` constructor:

https://github.com/bytefish/PgBulkInsert/blob/82b7b90613cb9b7eb0301ea0b4965c54918018da/PgBulkInsert/pgbulkinsert-jpa/src/main/java/de/bytefish/pgbulkinsert/jpa/mappings/PostgresTypeMapping.java#L23

As soon as the Maven repos work again, I will upload the 5.1.0-alpha1.

**Comment by bytefish** [comment]
Uploaded to Central Maven, it may take some time to propagate through their systems. Probably you can give it a try and let me know what you think about the API. Now off to work! 👍

**Comment by bytefish** [comment]
@t0bst4r I have released a 5.1.0, that includes the new API. If there are any problems or you have additional requests, feel free to reopen this ticket or create a new one. 😊
